### PR TITLE
Added check for invalid graph (exits if number of nodes is negative).

### PR DESCRIPTION
**Pull request by shivampdesai**
Use of gdb indicated that invalid graphs generated a negative number of nodes, which caused a segmentation fault. This check will display an error message if this is the case and terminate the program. 

This can be tested with the simple graph:
2 2 2
1 1
2 2

which used to seg fault but now will cause an exit out of the program.

### CODE REVIEW

**Comment by neoblizz** [comment]
Can one of the admins verify this patch?

**Comment by neoblizz** [comment]
@shivampdesai this addresses one of the issues, I think the big portion of this project was getting a system that recognizes if true MTX is passed to gunrock, and if not it should print out an error. Would you consider tackling this? https://github.com/gunrock/gunrock/issues/261

**Comment by shivampdesai** [comment]
Yes. What are the characteristics that would define a file in MTX format (things that I can check for)?

**Comment by neoblizz** [comment]
@shivampdesai it is all right here: http://math.nist.gov/MatrixMarket/formats.html

**Comment by shivampdesai** [comment]
free(coo) is done in both cases and I updated it so that it is less than 80 characters.

File: gunrock/graphio/market.cuh [hunk 1]
```diff
@@ -327,7 +327,8 @@ int ReadMarketStream(
 
 		if (ll_row <  1 || ll_col <  1)
 		{
-		    fprintf(stderr, "\nError Invalid graph: Bad MTX format, indices cannot start with 0.\n");
+		    fprintf(stderr, "\nError Invalid graph: Bad MTX format, indices
```
Comment (jowens) [inline gunrock/graphio/market.cuh]: Pretty sure you don't want a newline in the middle of a string. I think what you want is

```
fprintf(stderr, "start of text "
        "next line of text");
```

File: gunrock/graphio/utils.cuh [hunk 2]
```diff
@@ -65,8 +65,7 @@ SizeT RandomNode (SizeT num_nodes)
     return node_id % num_nodes;
 }
 
-template <typename VertexId, typename SizeT, typename Value>
-void RemoveStandaloneNodes(
+template <typename VertexId, typename SizeT, typename Value>void RemoveStandaloneNodes(
```
Comment (jowens) [inline gunrock/graphio/utils.cuh]: not sure why this change is here -- put a newline between `Value>` and `void`

File: gunrock/graphio/market.cuh [hunk 3]
```diff
@@ -332,6 +324,15 @@ int ReadMarketStream(
             {
                 num_input = sscanf(line, "%lld %lld %lf",
                     &ll_row, &ll_col, &lf_value);
+
+		if (ll_row <  1 || ll_col <  1)
+		{
+		    fprintf(stderr, "\nError Invalid graph: Bad MTX format, indices
+			    cannot start with 0.\n");
+		    if (coo) free(coo);
```
Comment (jowens) [inline gunrock/graphio/market.cuh]: I still see an `if (coo)` here ...

File: gunrock/graphio/market.cuh [hunk 4]
```diff
@@ -346,8 +347,7 @@ int ReadMarketStream(
 
                 else if (array || num_input < 2)
                 {
-                    fprintf(stderr,
-                            "Error parsing MARKET graph: badly formed edge\n");
+                    fprintf(stderr,                            "Error parsing MARKET graph: badly formed edge\n");
                     if (coo) free(coo);
```
Comment (jowens) [inline gunrock/graphio/market.cuh]: I still see an `if (coo)` here ...
